### PR TITLE
PointView

### PR DESCRIPTION
**Pull request by raphaelseher**
Screenshot from 2022-06-09 17-09-02
Screenshot from 2022-06-09 17-10-09

### CODE REVIEW

File: app/src/main/java/at/aau/se2/gwent/views/common/PointView.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  private void init(Context context) {
+    binding = ViewPointBinding.inflate(LayoutInflater.from(context));
+    addView(binding.getRoot());
+  }
+
+  public void bind(BoardViewData viewData) {
+    binding.mainButton.setText(viewData.getPrimaryButtonMode().getText());
+    binding.mainButton.setEnabled(viewData.isPrimaryButtonEnabled());
+
+    binding.turnTextView.setText(viewData.isMyTurn() ? "Your turn" : "Opponents turn");
```
Comment (Sikredo) [inline app/src/main/java/at/aau/se2/gwent/views/common/PointView.java:34]: Lets use a constant for "Your turn" and "Opponents turn"

File: app/src/main/java/at/aau/se2/gwent/views/common/PointView.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    binding.roundTextView.setText("Round " + viewData.getRoundNumber());
+    if (viewData.getCurrentGameState() == GameState.WAIT_FOR_OPPONENT) {
+      binding.turnTextView.setText("Waiting for opponent");
```
Comment (Sikredo) [inline app/src/main/java/at/aau/se2/gwent/views/common/PointView.java:37]: Use also a constant string here

File: app/src/main/java/at/aau/se2/gwent/views/common/PointView.java (hunk 1 above)
Comment (Sikredo) [inline app/src/main/java/at/aau/se2/gwent/views/common/PointView.java:34]: Lets use a constant here for "Your turn" and "Opponents turn"

File: app/src/main/java/at/aau/se2/gwent/views/common/PointView.java (hunk 2 above)
Comment (Sikredo) [inline app/src/main/java/at/aau/se2/gwent/views/common/PointView.java:37]: Also use a String constant here

File: app/src/main/java/at/aau/se2/gwent/views/startMenu/StartFragment.java [hunk 3]
```diff
@@ -125,39 +120,35 @@ public void onMultiWindowModeChanged(boolean isInMultiWindowMode) {
...
-                              break;
-                          }
-                        });
-                  }
-                })
-            .setNegativeButton("Cancel", null)
-            .create();
-    dialog.show();
+    new MaterialAlertDialogBuilder(context)
+        .setTitle("Join Game")
```
Comment (Sikredo) [inline app/src/main/java/at/aau/se2/gwent/views/startMenu/StartFragment.java:124]: String constant

File: app/src/main/java/at/aau/se2/gwent/views/startMenu/StartFragment.java [hunk 4]
```diff
@@ -125,39 +120,35 @@ public void onMultiWindowModeChanged(boolean isInMultiWindowMode) {
...
+        .setMessage("Input GameId from other Player.")
```
Comment (Sikredo) [inline app/src/main/java/at/aau/se2/gwent/views/startMenu/StartFragment.java:125]: String constant

File: gamelogic/src/main/java/at/aau/se2/gamelogic/models/GameField.java [hunk 5]
```diff
@@ -48,31 +48,29 @@ public int getRoundNumber() {
...
-    return (getPointsForPlayer(currentPlayer) > getPointsForPlayer(opponent))
-        ? currentPlayer
-        : opponent;
+    return (currentPoints > opponentPoints) ? currentPlayer : opponent;
   }
 
   @Exclude
-  public int getPointsForPlayer(Player player) {
+  public int getPointsForPlayer(InitialPlayer player) {
     if (player == null) return 0;
```
Comment (Sikredo) [inline gamelogic/src/main/java/at/aau/se2/gamelogic/models/GameField.java:63]: Should we here return -1 as this is kind of an Error case?

**Review by Sikredo — CHANGES_REQUESTED** [review]

File: gamelogic/src/main/java/at/aau/se2/gamelogic/models/GameField.java (hunk 5 above)
Comment (raphaelseher) [reply gamelogic/src/main/java/at/aau/se2/gamelogic/models/GameField.java:63]: Default fürs UI ist eh 0 und im Fall des Falles, kann man auf dem Gamestate zurückgreifen und was anderes als 0 anzeigen.

**Comment by raphaelseher** [comment]
|File|Coverage [11.08%]|:green_apple:|
|:-|:-:|:-:|
|[GameField.java](https://github.com/SE2-Gwent/Gwent/blob/a632059369377e372774cf504a5b7c2afe6652d0/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameField.java)|82.89%|:green_apple:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/a632059369377e372774cf504a5b7c2afe6652d0/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2FstartMenu%2FStartFragment.java)|0%|:green_apple:|
|[GameDebugViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/a632059369377e372774cf504a5b7c2afe6652d0/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdebug%2FGameDebugViewModel.java)|0%|:green_apple:|
|[GameDebugFragment.java](https://github.com/SE2-Gwent/Gwent/blob/a632059369377e372774cf504a5b7c2afe6652d0/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdebug%2FGameDebugFragment.java)|0%|:green_apple:|
|[BoardViewData.java](https://github.com/SE2-Gwent/Gwent/blob/a632059369377e372774cf504a5b7c2afe6652d0/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewData.java)|0%|:green_apple:|
|[BoardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/a632059369377e372774cf504a5b7c2afe6652d0/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardFragment.java)|0%|:green_apple:|
|[DetailedCardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/a632059369377e372774cf504a5b7c2afe6652d0/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FDetailedCardFragment.java)|0%|:green_apple:|
|[PointView.java](https://github.com/SE2-Gwent/Gwent/blob/a632059369377e372774cf504a5b7c2afe6652d0/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fcommon%2FPointView.java)|0%|:green_apple:|

|Total Project Coverage|22.88%|:green_apple:|
|:-|:-:|:-:|

**Pull request merged**